lib/presentation/engage/bible_series/widgets/image_body.dart

```diff
@@ -164,58 +163,72 @@ class _ImageInputBodyState extends State<ImageInputBodyState> {
+                            height: MediaQuery.of(context).size.width * 1.6,
+                            child: PhotoView(
+                              backgroundDecoration:
+                                  BoxDecoration(color: Colors.transparent),
+                              imageProvider: NetworkImage(state
+                                      .downloadURL[widget.contentNum.toString()]
+                                  [0]),
+                              minScale: PhotoViewComputedScale.contained * 0.8,
+                              maxScale: PhotoViewComputedScale.covered * 2,
+                              loadingBuilder: (context, event) => Center(
+                                child: CircularProgressIndicator(),
+                              ),
                             ),
                           ),
                         ),
-                      ),
-                      Positioned(
-                        left: 30,
-                        top: 30,
-                        child: GestureDetector(
-                          onTap: () {
-                            Navigator.of(context, rootNavigator: true).pop();
-                          },
-                          child: Center(
-                            child: Icon(
-                              Icons.close,
-                              size: 30,
-                              color: Colors.white,
+                        Positioned(
+                          left: 30,
+                          top: 30,
+                          child: GestureDetector(
+                            onTap: () {
+                              Navigator.of(context, rootNavigator: true).pop();
+                            },
+                            child: Center(
+                              child: Icon(
+                                Icons.close,
+                                size: 30,
+                                color: Colors.white,
+                              ),
                             ),
                           ),
                         ),
-                      ),
-                    ]),
-                  );
-                });
-          },
-          child: Container(
-            child: Image.network(downloadURL),
-          ),
-        ),
+                      ]),
+                    );
+                  });
+            },
+            child: Container(
+                width: MediaQuery.of(context).size.width * 0.8,
+                child: imageWidget(state, contentNum.toString()))),
       ],
     );
   }
 }
+
+Widget imageWidget(CompletionsState state, String contentNum) {
+  if (state.thumbnailURL != null && state.thumbnailURL[contentNum] != null) {
+    return Image.network(
+      state.thumbnailURL[contentNum][0],
+      fit: BoxFit.fill,
+    );
+  } else if (state.localImage != null && state.localImage[contentNum] != null) {
+    return Image.file(state.localImage[contentNum][0]);
```

Comment:
Do we need the BoxFit.fill here as well?

Reply:
yeah it enlarges the images no matter its pixel resolution

Reply:
so all the thumbnail and image would be the same size

Reply:
So do we need it here? I noticed that we have BoxFill for the network image but not for the file image.

Reply:
for file image, the image is larger than the container size, so it gets restricted. For thumbnails, it is smaller so a BoxFill is needed to stretch it to however large the container is